crates/ruff_python_parser/src/semantic_errors.rs

```diff
@@ -646,17 +657,56 @@ impl SemanticSyntaxChecker {
                     }
                 }
             }
-            Expr::Yield(ast::ExprYield {
-                value: Some(value), ..
-            }) => {
-                // test_err single_star_yield
-                // def f(): yield *x
-                Self::invalid_star_expression(value, ctx);
+            Expr::Yield(ast::ExprYield { value, .. }) => {
+                if let Some(value) = value {
+                    // test_err single_star_yield
+                    // def f(): yield *x
+                    Self::invalid_star_expression(value, ctx);
+                }
+                self.yield_outside_function(ctx, expr, YieldOutsideFunctionKind::Yield);
+            }
+            Expr::YieldFrom(_) => {
+                self.yield_outside_function(ctx, expr, YieldOutsideFunctionKind::YieldFrom);
+            }
+            Expr::Await(_) => {
+                self.yield_outside_function(ctx, expr, YieldOutsideFunctionKind::Await);
             }
             _ => {}
```

Comment:
Are lambdas consider to be in a function scope or class scope or do they not change the scope?

For example, the following (in module scope) shouldn't raise an error)

```
a = lambda a: yield a
```